hwilib/devices/trezor.py

```diff
@@ -95,6 +98,38 @@ def parse_multisig(script):
     multisig = proto.MultisigRedeemScriptType(m=m, signatures=[b''] * n, pubkeys=pubkeys)
     return (True, multisig)
 
+# Parses the PSBT_GLOBAL_XPUB fields of a PSBT as multisig pubkeys
+def parse_multisig_xpubs(tx, psbt_in_out, multisig):
+    try:
+        old_pubs = [k.node.public_key for k in multisig.pubkeys]
+        if len(old_pubs) != sum([len(xpubs) for xpubs in tx.global_xpubs.values()]):
```

Comment:
I'm not sure that this check is necessary. There could be more xpubs given than in the particular multisig being examined.